main.cpp

```diff
@@ -1,478 +1,585 @@
-    These 5 sub-objects will not be defined using Primitives, but instead will be their own UDTs 
-    you'll define these 5 sub-objects in Part 1d.
+1) Audio Interface
+5 properties:
+    1) brand name (std::string)
+    2) number of inputs (int)
+    3) number of outputs (int)
+    4) has an ability to provide phantom power for each input (bool)
+    5) has a volume knob (bool)
+3 things it can do:
+    1) convert analog signal from a microphone in digital signal
+    2) send signal from input to DAW
+    3) change the volume of the output signal
 */
 
+struct AudioInterface
+{
+    //1) brand name (std::string)
+    std::string brandName = "Native Instruments";
+    //2) number of inputs (int)
+    int numberOfInputs = 2;
+    //3) number of outputs (int)
+    int numberOfOutputs = 2;
+    //4) has an ability to provide phantom power for each input (bool)
+    bool abilityToProvidePhantomPowerForEachInput = true;
+    //5 has a volume knob (bool)
+    bool hasAVolumeKnob = true;
+
+    //1) convert analog signal from a microphone in digital signal
+    void convertAnalogToDigitalSignal();
+    //2) send signal from input to DAW
+    void sendToDAW();
+    //3) change the volume of the output signal
+    void changeOutputVolume();
+};
+
 /*
-Thing 10)
+2) Automatic coffee machine
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) has a milk frother (bool)
+    2) number of drinks types that can be prepared (int)
+    3) number of programmed coffee strength (int)
+    4) has an automatic milk system cleaning program (bool)
+    5) number of drinks that can be prepared in a minute (double)
 3 things it can do:
-    1)
-    2)
-    3)
- */
+    1) grind coffee beans
+    2) froth the milk
+    3) make a cappuccino
+*/
+
+struct AutomaticCoffeeMachine
+{
+    //1) has a milk frother (bool)
+    bool hasMilkFrother = true;
+    //2) number of drinks types that can be prepared (int)
+    int numberOfDrinksTypes = 5;
+    //3) number of programmed coffee strength (int)
+    int numberOfProgrammedCoffeeStrength = 3;
+    //4) has an automatic milk system cleaning program (bool)
+    bool hasAutomaticMilkSystemCleaningProgram = true;
+    //5) number of drinks that can be prepared in a minute (double)
+    double numberOfDrinksPreparedPerMinute = 2.5;
+
+    struct CoffeeBeans
+    {
+        //1) type of coffee beans (std::string)
+    std::string type = "arabica";
+        //2) type of coffee based on the origin (std::string)
+    std::string typeBasedOnTheOrigin = "blend";
+        //3) growing region (std::string)
+    std::string growingRegion = "Ethiopia";
+        //4) type of roasts (std::string)
+    std::string typeOfRoasts = "dark";
+        //5) content of  caffeine per cup [mg] (int)
+    int contentOfCaffeinePerCupMl = 100;
+
+        //1) import coffe beans
+    void import();
+        //2) check the moisture of coffe beans
+    void checkTheMoisture();
+        //2) roast coffee beans
+    void roast();
+
+
+    };
```

Comment:
Code inside of curly braces should be indented 4 spaces relative to the curly braces
- [ ] revise